pygments/lexers/shell.py

```diff
@@ -772,7 +772,7 @@ class PowerShellSessionLexer(ShellSessionBaseLexer):
     mimetypes = []
 
     _innerLexerCls = PowerShellLexer
-    _ps1rgx = re.compile(r'^(PS [^>]+> )(.*\n?)')
+    _ps1rgx = re.compile(r'^((?:\[[^]]+\]: )?PS [^>]+> ?)(.*\n?)')
```

Comment:
Sorry for the delay, I didn't have time to spare on this until now.
This second question mark was actually to make the block in square brackets optional, allowing the regex to match both local and remote PowerShell sessions, without having to create a separate subclass for the remote case.

Reply:
Sorry, I wasn't clear :( I was referring to the one *after* the `>`, i.e. this one: 
```
^((?:\[[^]]+\]: )?PS [^>]+> ?)(.*\n?)
                            ^
```
Previously, that wasn't there. I understand why you want to make the first block optional, but that one I didn't quite get (and I just realized it's the _third_ `?`, not the second ...)

Reply:
Oh right, if I remember correctly it was due to the fact that sometimes the space after the `>` isn't printed, so I figured I'd make it optional. It might have just been an error in my tests though. Not sure about policies, but I'd say it doesn't hurt to keep it in there right?

Reply:
I guess so :)